### PR TITLE
Update Firewall v3 MIM

### PR DESCRIPTION
**Pull request by robertschaedler3**
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/mim/firewall.json [hunk 1]
```diff
@@ -12,19 +12,19 @@
           "desired": false,
           "schema": {
             "type": "enum",
-            "valueSchema": "integer",
+            "valueSchema": "string",
             "enumValues": [
               {
                 "name": "unknown",
-                "enumValue": 0
+                "enumValue": "unknown"
```
Comment (MariusNi) [inline src/modules/mim/firewall.json:19]: >unknown [](http://example.com/codeflow?start=30&length=7)

"none" would be more universal than "unknown". Here and elsewhere

File: src/modules/mim/firewall.json [hunk 2]
```diff
@@ -36,102 +36,259 @@
...
+          "schema": {
+            "type": "enum",
+            "valueSchema": "string",
+            "enumValues": [
+              {
+                "name": "unknown",
+                "enumValue": "unknown"
+              },
+              {
+                "name": "success",
```
Comment (MariusNi) [inline src/modules/mim/firewall.json:51]: >success [](http://example.com/codeflow?start=25&length=7)

This is a state, so "succeeded" and "failed". 
Or: change "configuredState" to "configurationStatus" and keep these

File: src/modules/mim/firewall.json [hunk 3]
```diff
@@ -36,102 +36,259 @@
...
+        {
+          "name": "defaultPolicies",
+          "type": "mimObject",
+          "desired": false,
           "schema": {
             "type": "object",
             "fields": [
               {
-                "name": "action",
+                "name": "in",
```
Comment (MariusNi) [inline src/modules/mim/firewall.json:75]: >in [](http://example.com/codeflow?start=25&length=2)

Too short. "input"?

File: src/modules/mim/firewall.json [hunk 4]
```diff
@@ -36,102 +36,259 @@
...
+              "type": "object",
+              "fields": [
+                {
+                  "name": "desiredState",
+                  "schema": {
+                    "type": "enum",
+                    "valueSchema": "string",
+                    "enumValues": [
+                      {
+                        "name": "exist",
```
Comment (MariusNi) [inline src/modules/mim/firewall.json:143]: >exist [](http://example.com/codeflow?start=33&length=5)

Better:

"present", "missing"

File: src/modules/mim/firewall.json [hunk 5]
```diff
@@ -36,102 +36,259 @@
...
+                    {
+                      "name": "reject",
+                      "enumValue": "reject"
+                    }
+                  ]
+                }
               },
               {
-                "name": "protocol",
+                "name": "out",
```
Comment (MariusNi) [inline src/modules/mim/firewall.json:269]: >out [](http://example.com/codeflow?start=25&length=3)

We need better names for these. Out what? Same for in.

File: src/modules/mim/firewall.json [hunk 6]
```diff
@@ -36,102 +36,259 @@
...
+                },
+                {
+                  "name": "protocol",
+                  "schema": {
+                    "type": "enum",
+                    "valueSchema": "string",
+                    "enumValues": [
+                      {
+                        "name": "any",
+                        "enumValue": "any"
```
[the remark this answers is not in the document]
Comment (robertschaedler3) [reply src/modules/mim/firewall.json:199]: In this case specifying `none` would be the same as `any`. In both cases the resulting rule specification will not include a protocol (omitting protocol `-p <protocol>` implies *any*).

File: src/modules/mim/firewall.json [hunk 7]
```diff
@@ -36,105 +36,252 @@
...
-                "name": "sourceAddress",
-                "schema": "string"
-              },
+            "type": "enum",
+            "valueSchema": "string",
+            "enumValues": [
               {
-                "name": "sourcePort",
-                "schema": "integer"
+                "name": "unknown",
```
Comment (MariusNi) [inline src/modules/mim/firewall.json:47]: >unknown [](http://example.com/codeflow?start=25&length=7)

none

File: src/modules/mim/firewall.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (MariusNi) [reply src/modules/mim/firewall.json:194]: No, "none" is never same as any defined value such as "any". Let's add "none" for 0.

File: src/modules/mim/firewall.json (hunk 7 above)
Comment (MariusNi) [reply src/modules/mim/firewall.json]: As discussed offline

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**